elasticapm/traces.py

```diff
@@ -325,6 +328,8 @@ def to_dict(self):
             "sampled": self.is_sampled,
             "span_count": {"started": self._span_counter - self.dropped_spans, "dropped": self.dropped_spans},
         }
+        if self.sample_rate is not None:
+            result["sample_rate"] = float(self.sample_rate) if self.sample_rate != "0" else 0
```

Comment:
In the [spec](https://github.com/elastic/apm/blob/master/specs/agents/tracing-sampling.md#non-sampled-transactions), it explicitly states `0` as opposed to `0.0` (which is what it would be if it were formatted as a float).

In retrospect, I highly doubt it matters. I'm going to omit the special handling and let it be sent as a `0.0` float. Do you anticipate any problems with that @axw ?

Reply:
Nope, that's fine, I think it's reasonable to expect agents to parse `0.0` the same as `0`.

It would be nice to send as few bytes as necessary, but 2 bytes isn't going to matter in the grand scheme of things.